src/fileimpl.cpp

```diff
@@ -407,24 +408,10 @@ offset_t readOffset(const Reader& reader, size_t idx)
   article_index_t FileImpl::getNamespaceEndOffset(char ch)
   {
     log_trace("getNamespaceEndOffset(" << ch << ')');
+    ASSERT(ch, >=, 32);
+    ASSERT(ch, <, 127);
```

Comment:
Shouldn't we accept getting the end offset of namespace 127 ?

Not sure we need it (`char(127)` is DEL char), but if we accept getting the start offset, we should probably accept getting end offset also.

In fact I wonder why we limit to ascii char. We limit ourselves to ascii char but the code is not limited to that. Any value lower that 256 should work.

I'm just questionning. I will accept that we limit namespace values and that we don't.

Reply:
> if we accept getting the start offset, we should probably accept getting end offset also.

`getNamespaceEndOffset(ch)` is defined as `getNamespaceBeginOffset(ch+1)`, therefore the upper limit on `ch` in  `getNamespaceEndOffset()` must be one less than that in `getNamespaceBeginOffset()`.

> In fact I wonder why we limit to ascii char. We limit ourselves to ascii char but the code is not limited to that. Any value lower that 256 should work.

A minor problem is that `char` is a different type from `unsigned char` (and can be a signed type, depending on implementation). Of course that can be worked around easily in different ways, but using a printable subset for namespaces is a safer solution for several reasons (one example is not having to deal with UTF-8 or other encoding issues, if we ever need to convert the `{namespace, path}` to a full url of the form 'n/path').

Reply:
> `getNamespaceEndOffset(ch)` is defined as `getNamespaceBeginOffset(ch+1)`

`getNamespaceEndOffset` of the last namespace exists and is defined to the size of the array.

> one example is not having to deal with UTF-8 or other encoding issues, if we ever need to convert the {namespace, path} to a full url of the form 'n/path'

Indeed, using printable char simplify a lot this. Let's limit ourselves to printable char.
But 127 is not a printable char, we should drop it. ('/' may cause problem also, but we will not avoid it here)

Reply:
> Shouldn't we accept getting the end offset of namespace 127 ?
>
> `getNamespaceEndOffset` of the last namespace exists and is defined to the size of the array.
>
> But 127 is not a printable char, we should drop it. 

The last allowed namespace is 126 (thus 127 is already dropped in `getNamespaceEndOffset()`). 127 is allowed in `getNamespaceBeginOffset()` since `getNamespaceEndOffset()` calls the former with an incremented argument.

Reply:
> The last allowed namespace is 126

This make sense. I would have simply use the size of the array to get the end offset of the last namespace but it is not so important. We will probably not use `~` as namespace and this is cached anyway.